// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package apps.proto;

import "chrome/browser/apps/almanac_api_client/proto/client_context.proto";

// This file is a mirror of the proto file maintained in the server code base at
// go/app-deduplication-service-proto. Changes should be made by updating the
// server code base and then copying the result to Chromium.

message DeduplicateRequest {
  // Context information about the device making this request.
  optional ClientDeviceContext device_context = 1;

  // Context information about the user making this request.
  optional ClientUserContext user_context = 2;
}

message DeduplicateResponse {
  // An Application Group representing a set of "duplicate" apps.
  message AppGroup {
    optional string app_group_uuid = 1;

    // Applications which belong to the App Group.
    repeated string package_id = 2;
  }

  // The list of Application Groups corresponding to the requested applications.
  repeated AppGroup app_group = 1;
}
